PSGSuite/Public/Users/Invoke-GSUserOffboarding.ps1

```diff
@@ -0,0 +1,158 @@
+                            $updateParams['Password'] = ConvertTo-SecureString (New-RandomPassword) -AsPlainText -Force
+                            $updateParams['ChangePasswordAtNextLogin'] = $true
+                        }
+                        '(Full|MoveToOrgUnit)' {
+                            if ($PSBoundParameters.ContainsKey('DestinationOrgUnit')) {
+                                $updateParams['OrgUnitPath'] = $PSBoundParameters['DestinationOrgUnit']
+                            }
+                            else {
+                                throw "No DestinationOrgUnit provided!! Stopping further processing"
+                                exit 1
+                            }
+                        }
+                    }
+                }
+                "[$(Get-Date -Format o)] [$U] Updating user"
+                Update-GSUser @_user @updateParams | Format-List PrimaryEmail,@{N = "FullName";E = {$_.name.fullName}},Suspended,ChangePasswordAtNextLogin,OrgUnitPath
+                if ($Options -contains 'Full' -or $Options -contains 'ClearASPs') {
+                    "[$(Get-Date -Format o)] [$U] Retrieving App Specific Passwords to remove"
+                    $ASPs = Get-GSUserASPList @_user
+                    if ($ASPs) {
+                        foreach ($ASP in $ASPs) {
+                            "[$(Get-Date -Format o)] [$U] Revoking ASP for '$($ASP.name)'"
+                            Remove-GSUserASP @_user -CodeID $ASP.codeId -Confirm:$false
+                        }
+                        Remove-Variable ASPs -ErrorAction SilentlyContinue
+                    }
+                    else {
+                        "[$(Get-Date -Format o)] [$U] User has no ASP's to remove!"
+                    }
+                }
+                if ($Options -contains 'Full' -or $Options -contains 'ClearOAuthTokens') {
+                    "[$(Get-Date -Format o)] [$U] Retrieving OAuth Tokens to remove"
+                    $Tokens = Get-GSUserTokenList @user
+                    if ($Tokens.clientId) {
+                        foreach ($Token in $Tokens) {
+                            "[$(Get-Date -Format o)] [$U] Revoking OAuth Token for '$($Token.displayText)'"
+                            Remove-GSUserToken @user -ClientID $Token.clientId -Confirm:$false
+                        }
+                        Remove-Variable Tokens -ErrorAction SilentlyContinue
+                    }
+                    else {
+                        "[$(Get-Date -Format o)] [$U] User has no OAuth Tokens to remove!"
+                    }
+                }
+                if ($Options -contains 'Full' -or $Options -contains 'RemoveMobileDevices') {
+                    "[$(Get-Date -Format o)] [$U] Retrieving Mobile Devices to remove"
+                    $Mobiles = Get-GSMobileDeviceList @user -Projection BASIC
+                    if ($Mobiles) {
+                        foreach ($Mobile in $Mobiles) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The cmdlet 'Invoke-GSUserOffboarding' returns an object of type 'System.String' but this type is not declared in the OutputType attribute.](https://app.codacy.com/manual/scrthq/PSGSuite/pullRequest?prid=4742353)